backend/FwLite/LcmCrdt/CurrentProjectService.cs

```diff
@@ -82,18 +83,27 @@ public async ValueTask<ProjectData> RefreshProjectData()
         return projectData;
     }
 
-    private async Task MigrateDb()
+    private static readonly ConcurrentDictionary<string, Lazy<Task>> MigrationTasks = [];
+    private Task MigrateDb()
     {
-        try
+        //ensure we only execute once, otherwise we'll have a conflict as Migrate is not thread safe.
+        //design based on https://andrewlock.net/making-getoradd-on-concurrentdictionary-thread-safe-using-lazy/
+#pragma warning disable VSTHRD011
+        return MigrationTasks.GetOrAdd(Project.DbPath, _ => new Lazy<Task>(Execute)).Value;
```

Comment:
As this will now only every happen once per project, we could remove the `if` condition we use for calling `MigrateDb`.
On the other hand, we might revert this at some point, because [ef9 is supposed to handle this for us](https://learn.microsoft.com/en-us/ef/core/providers/sqlite/limitations#concurrent-migrations-protection) (with something to watch out for when using SQLite) 🤷.

Either way, this code looks good to me and we probably shouldn't burn cycles on it if it's working.

Reply:
oh that's cool, I was a bit surprised that the DB didn't do this for us, but it makes sense that it should be done at the DB level not app level which does make it more complicated to implement.